packages/jsonc/jsonc.mjs

```diff
@@ -1,14 +1,12 @@
-import { getFilenameExt } from '@nodejs-loaders/parse-filename';
 import stripJsonComments from 'strip-json-comments';
 
 /**
  * @type {import('node:module').ResolveHook}
  */
 async function resolveJSONC(specifier, ctx, nextResolve) {
 	const nextResult = await nextResolve(specifier);
-	const ext = getFilenameExt(nextResult.url);
 
-	if (ext === '.jsonc' && ctx.importAttributes.type === 'jsonc')
```

Comment:
I think import attrs may not be required because node can't handle this anyway (so it won't know to barf), so you still need to check file extension when import attrs is not there.

```suggestion
	const ext = getFilenameExt(nextResult.url);

	if (
		ctx.importAttributes?.type === 'jsonc'
		|| (
			!ctx.importAttributes?.type
			&& ext === '.jsonc'
		)
	)
```